lib/automation/report_runner.rb

```diff
@@ -153,6 +155,10 @@ def run!
           top_p: @top_p,
           user: Discourse.system_user,
           feature_name: "ai_report",
+          feature_context: {
+            automation_id: @automation&.id,
+            automation_name: @automation&.name,
+          },
```

Comment:
Minor but maybe we `feature_context` should be empty hash and only merge ` automation_id` and `automation_name` when `@automation` exists?

Reply:
I am so on the fence here ... automation being nil is kind of a usage bug so the odds of this happening is just tiny.